### PR TITLE
[TASKMGR] Fix number formatting to respect NLS Number format settings

### PR DESCRIPTION
**Pull request by KRosUser**
## Purpose

TaskMgr currently does not respect local NLS settings and forces 3 digits + comma (US convention) formatting

image

## Proposed changes

Use user defined formatting preference

image

Implementation with a direct reuse of SH_FormatInteger as already implemented in other places ( dll/win32/shell32/dialogs/filedefext.cpp ... )

By opportunity change the design pattern to switch/case as suggested by @katahiromz

### CODE REVIEW

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:328]:
```suggestion
SH_FormatInteger(
    _In_ LONGLONG Num,
    _Out_ LPWSTR pwszResult,
    _In_ UINT cchResultMax)
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:688]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:694]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:700]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:706]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:712]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:718]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:332]:
```suggestion
    StringCchPrintfW(wszNumber, _countof(wszNumber), L"%I64u", Num);
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:327]:
```suggestion
/*
 * Adapted from SH_FormatInteger in dll/win32/shell32/dialogs/filedefext.cpp.
 */
UINT
```

File: base/applications/taskmgr/precomp.h
Comment (HBelusca) [inline base/applications/taskmgr/precomp.h:22]: Move in-between wingdi and winuser.

**Comment by KRosUser** [comment]
@HBelusca : We're all set !

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:637]: It would be better to use `switch`-`case`.

Comment (KRosUser) [reply base/applications/taskmgr/procpage.c:637]: It's not the purpose of this PR, but you're right. I'll change the design accordingly

Comment (katahiromz) [reply base/applications/taskmgr/procpage.c:637]: for later.

Comment (KRosUser) [reply base/applications/taskmgr/procpage.c:637]: Yes, implementation under test. Commit to follow.

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:618]:
```suggestion
    switch (ColumnDataHints[ColumnIndex])
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:625]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetProcessId(Index));
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:637]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetSessionId(Index));
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:641]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%02lu", PerfDataGetCPUUsage(Index));
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:647]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:657]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:662]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:680]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:675]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:685]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:689]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetBasePriority(Index));
```

File: base/applications/taskmgr/procpage.c
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:652]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

**Comment by KRosUser** [comment]
@HBelusca : I let you review following changed introduced after comments from @katahiromz and approve/merge if OK for you (pending checks are finally OK).

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:383]:
```suggestion
    cchResult = GetNumberFormatW(LOCALE_USER_DEFAULT,
                                 0,
                                 wszNumber,
                                 &nf,
                                 pwszResult,
                                 cchResultMax);
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:644]:
```suggestion
        case COLUMN_CPUTIME:
        {
            LARGE_INTEGER time;
            DWORD dwHours;
            DWORD dwMinutes;
            DWORD dwSeconds;

```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:616]:
```suggestion
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:648]:
```suggestion
            break;
        }
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:739]:
```suggestion
    return TRUE;
```

**Review by HBelusca — APPROVED** [review]

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:332]:
```suggestion
    _Out_writes_z_(cchResultMax) LPWSTR pwszResult,
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:617]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:621]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:625]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:629]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:633]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:637]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:649]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:655]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:660]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:665]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:669]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:673]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:678]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:683]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:688]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:692]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:696]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:700]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:704]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:708]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:713]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:718]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:723]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:728]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:733]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:738]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:741]:
```suggestion
    return FALSE;
```

**Pull request merged**
